terraform/modules/emrfs-lambdas/cognito-rds-sync-lambda-files/aws_sync_caller.py

```diff
@@ -0,0 +1,54 @@
+import boto3
+
+rds_data_client = boto3.client('rds-data')
+sts_connection = boto3.client('sts')
+
+# assumes mgmt role to set up cognito client associated with mgmt account
+def create_cognito_client(mgmt_account_role_arn):
+    mgmt_account = sts_connection.assume_role(
+        RoleArn=mgmt_account_role_arn,
+        RoleSessionName="mgmt_cognito_rds_sync_lambda"
+    )
+    # create cognito client using the assumed role credentials in mgmt acc
+    global cognito_client
+    cognito_client = boto3.client(
+        'cognito-idp',
+        aws_access_key_id=mgmt_account['Credentials']['AccessKeyId'],
+        aws_secret_access_key=mgmt_account['Credentials']['SecretAccessKey'],
+        aws_session_token=mgmt_account['Credentials']['SessionToken']
+    )
+
+
+# returns list of all users and their 'sub' attribute from userpool - includes enabled status
+def get_users_in_userpool(user_pool_id):
```

Comment:
I'm a bit concerned we are not asking for preferred_username here also, since its needed to workout the username for ADFS users...